### ISSUE TITLE
#39 is back in v0.6.0

### ISSUE THREAD

**Issue by szvincze** [spiffe/spire-controller-manager#453]
It seems that the solution of https://github.com/spiffe/spire-controller-manager/issues/39 is missing from v0.6.0 therefore there is a restart because the spire-server socket is not ready.

**Comment by faisal-memon** [comment]
It got removed with https://github.com/spiffe/spire-controller-manager/pull/418, bumping the controller-runtime version

**Comment by szvincze** [comment]
> It got removed with #418, bumping the controller-runtime version

Thanks, @faisal-memon.

I have seen it got removed, but this way it causes unwanted container restart again. Do you plan to include it again?

**Comment by faisal-memon** [comment]
@szvincze Yes we should add this functionality back in. The API has changed and there is a few different ways to go about it. Will update with a few different options.

**Comment by szvincze** [comment]
@faisal-memon Thanks, I am looking forward to the mentioned options.
